MIGRATING.md

```diff
@@ -1,8 +1,85 @@
 [AuthState]: https://github.com/okta/okta-auth-js#authstatemanager
 [transformAuthState]: https://github.com/okta/okta-auth-js/blob/master/README.md#transformauthstate
+[OktaAuth]: https://github.com/okta/okta-auth-js
 
 # Migrating
 
+## From version 3.x to 4.x
+
+### Updating `OktaConfig`
+
+From version 4.0, an `OktaConfig` starts to explicitly accept [OktaAuth][] instance to replace the internal `OktaAuthService`. You will need to replace the [OktaAuth related configurations](https://github.com/okta/okta-auth-js#configuration-reference) with a pre-initialized [OktaAuth][] instance.
+
+**Note**
+
+- `@okta/okta-auth-js` is now a peer dependency for this SDK. You must add `@okta/okta-auth-js` as a dependency to your project and install it separately from `@okta/okta-angular`.
```

Comment:
Why do we need to require an external oktaAuth instance? Couldn't we create an instance using our preferred version if none is provided? Seems better if installing, configuring and injecting auth-js separately is optional.

Reply:
It's the pattern we follow in `okta-react` and `okta-vue`. The reason we want to make the oktaAuth as a required config is because the default behaviour of these downstream SDKs start services of `oktaAuth` instance, then we do want to suggest devs to stick with only one oktaAuth instance to avoid unpredictable race conditions.